sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -52,7 +52,7 @@ singleCompoundStatement
     ;
 
 beginEndCompoundBlock
-    : BEGIN compoundBody END
+    : beginLabel? BEGIN compoundBody END endLabel?
```

Comment:
does the end label must be the same with begin label? If so why do we allow users to specify it?

Reply:
Standard allows user to specify labels at the beginning and at the end of the block. We check if labels are the same while building syntax tree.